### PR TITLE
Implement filter for Project Spaces

### PR DESCRIPTION
**Pull request by lookacat**
## Description
We've added the option to search for spaces in the project space overview.
See https://github.com/owncloud/web/issues/9650

## Related Issue

- Fixes https://github.com/owncloud/web/issues/9650

## Screenshots (if appropriate):
Screenshot (56)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/tests/unit/views/spaces/Projects.spec.ts [hunk 1]
```diff
@@ -75,6 +82,8 @@ describe('Projects view', () => {
 })
 
 function getMountedWrapper({ mocks = {}, spaces = [], abilities = [], stubAppBar = true } = {}) {
+  jest.mocked(queryItemAsString).mockImplementationOnce(() => '1')
```
Comment (lookacat) [inline packages/web-app-files/tests/unit/views/spaces/Projects.spec.ts:85]: is needed for pagination

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 2]
```diff
@@ -175,17 +197,23 @@ export default defineComponent({
...
     const store = useStore()
+    const router = useRouter()
+    const route = useRoute()
     const clientService = useClientService()
     const { selectedResourcesIds } = useSelectedResources({ store })
     const { can } = useAbility()
     const { current: currentLanguage, $gettext } = useGettext()
+    const filterTerm = ref('')
+    const markInstance = ref(undefined)
+    const tilesTableRef = ref(undefined)
```
Comment (JammingBen) [inline packages/web-app-files/src/views/spaces/Projects.vue:216]: Do you need the ref here? I don't see you using it.

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 3]
```diff
@@ -213,6 +241,44 @@ export default defineComponent({
       items: runtimeSpaces,
       fields: sortFields
     })
+    const filter = (spaces, filterTerm) => {
```
Comment (JammingBen) [inline packages/web-app-files/src/views/spaces/Projects.vue:244]: Add types to the args please 🙂

File: packages/web-app-files/src/views/spaces/Projects.vue (hunk 2 above)
Comment (lookacat) [reply packages/web-app-files/src/views/spaces/Projects.vue:223]: which ref are you talking about? I use all 3

Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/Projects.vue:223]: I mean `tilesTableRef`. You set in in the template, but I don't see that you actually using it... or am I missing something? 🤔

Comment (lookacat) [reply packages/web-app-files/src/views/spaces/Projects.vue:223]: i use it in the template, but yeah you are right i switched to a selector

**Review by JammingBen — APPROVED** [review]
Nice 👍

**Pull request merged**